subsys/modbus/modbus_server.c

```diff
@@ -926,6 +926,36 @@ static bool mbs_fc16_hregs_write(struct modbus_context *ctx)
 	return true;
 }
 
+static bool mbs_try_user_fc(struct modbus_context *ctx, uint8_t fc)
+{
+	struct modbus_custom_fc *p;
+
+	LOG_DBG("Searching for custom Modbus handlers for code %u", fc);
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&ctx->user_defined_cbs, p, node) {
+		if (p->fc == fc) {
+			LOG_DBG("Found custom handler");
+			int iface = modbus_iface_get_by_ctx(ctx);
+
+			p->excep_code = MODBUS_EXC_NONE;
+			bool rval = p->cb(iface, &ctx->rx_adu, &ctx->tx_adu, &p->excep_code,
+					p->user_data);
+
+			if (p->excep_code != MODBUS_EXC_NONE) {
```

Comment:
I have pointed this out a few times, please move the variable definition to the top of a block.
```c
	if (p->fc == fc) {
			int iface = modbus_iface_get_by_ctx(ctx);
			bool rval;

			LOG_DBG("Found custom handler");
			p->excep_code = MODBUS_EXC_NONE;
			rval = p->cb(iface, &ctx->rx_adu, &ctx->tx_adu, &p->excep_code,
					p->user_data);
```